benchmark/_interfaces.py

```diff
@@ -87,3 +87,25 @@ def measure(f, *a, **kw):
         :return: Deferred firing when measurement has been taken, with the
             value of the measurement.
         """
+
+
+class IRequestGenerator(Interface):
+    """
+    A request generator that provides a method to generate
+    a request of certain type (like read or write)
+    """
+    def make_request():
+        """
+        Interface for request generator
+        """
+
+
+class IScenarioSetup(Interface):
```

Comment:
Given each of these classes contains one method, I'd say either:
1. abandon the use of interfaces, and just pass the two functions separately.
2. combine these into a single interface with `request_setup` and `make_request` methods.  I note that your only non-trivial implementation combines both interfaces into a single implementing class.